src/main/java/com/ait/lienzo/client/core/shape/wires/AlignAndDistribute.java

```diff
@@ -190,16 +190,21 @@ public AlignAndDistributeControl getShapeControl(IPrimitive<?> prim)
         return m_shapes.get(prim.uuid());
     }
 
-    public AlignAndDistributeControl addShape(IDrawable<?> group)
-    {
+    public AlignAndDistributeControl addShape(IDrawable<?> group) {
+        return addShape(group, true);
+    }
+
+    public AlignAndDistributeControl addShape(IDrawable<?> group, boolean useChangeListener) {
         final String uuid = group.uuid();
 
         AlignAndDistributeControl handler = m_shapes.get(uuid);
 
         // only add if the group has not already been added
-        if (null == handler)
-        {
-            List<Attribute> attrs = group.getBoundingBoxAttributes();
+        if (null == handler) {
+            List<Attribute> attrs = null;
+            if (useChangeListener) {
```

Comment:
sorry @Josephblt I'm not following you by adding this `useChangeListener` flag. What's it intended for?

Reply:
Hey @romartin. This is a feature introduced by @mdproctor. Listening to all attribute changes can be quite demanding performance wise. This flag allows turning this on-off easily. It is set to false by default. To always listen to changes fixed the original problem but could impact on performance. So we decided to roll back the fix and make punctual adjustments to address this.

Reply:
Yeah but anyway not still sure if the code  is fine.. I mean, in case `useChangeListener` is set to `false`, the `attrs` will be `null`,  so does this makes sense? Although I understand right now it's set to `true`, does that will work in case on a near future we decide to change it? Thx for clarifying :+1: